File: app/src/main/java/com/jermaine/awesomelibraries/view/activity/base/BaseActivity.java
package com.jermaine.awesomelibraries.view.activity.base;


import android.support.v7.app.AppCompatActivity;

import com.jermaine.awesomelibraries.app.App;
import com.jermaine.awesomelibraries.di.component.AppComponent;

public class BaseActivity extends AppCompatActivity {
    public AppComponent getComponent() {
        return ((App) getApplication()).getComponent();
    }
}

File: app/src/main/java/com/jermaine/awesomelibraries/uicontroller/activity/base/BaseActivity.kt
package com.jermaine.awesomelibraries.uicontroller.activity.base


import android.arch.lifecycle.LifecycleActivity
import com.jermaine.awesomelibraries.app.App
import com.jermaine.awesomelibraries.di.component.AppComponent

open class BaseActivity : LifecycleActivity() {
    val component: AppComponent
        get() = (application as App).component
}
